src/audio/mux/mux.c

```diff
@@ -530,15 +530,18 @@ static int mux_process(struct processing_module *mod,
 	int frames;
 	int sink_bytes;
 	int source_bytes;
-	int i;
+	int i, j;
 
 	comp_dbg(dev, "mux_process()");
 
 	/* align source streams with their respective configurations */
+	j = 0;
+	frames = INT32_MAX;
 	list_for_item(clist, &dev->bsource_list) {
 		source = container_of(clist, struct comp_buffer, sink_list);
 		source_c = buffer_acquire(source);
 		if (source_c->source->state == dev->state) {
+			frames = MIN(frames, input_buffers[j].size);
```

Comment:
@singalsu do you think we should move this check for active state to the module adapter copy and only have active source buffers in the input_buffers? then we can simply iterate over the list of input_buffers[i].size to find the MIN and set the size for all input buffers after processing as well

Reply:
intention was combine line 570 - 579?

Reply:
I think it's a good idea and would help the clients. Could we do such fix later since it impacts many components and use this way as quick fix?